core/modules/startup/startup.js

```diff
@@ -30,7 +30,7 @@ exports.startup = function() {
 		// 2023-07-21 Edge returns UA below. So we use "isChromeLike"
 		//'mozilla/5.0 (windows nt 10.0; win64; x64) applewebkit/537.36 (khtml, like gecko) chrome/114.0.0.0 safari/537.36 edg/114.0.1823.82'
 		$tw.browser.isChromeLike = navigator.userAgent.toLowerCase().indexOf("chrome") > -1;
-		$tw.browser.hasTouch = window.matchMedia("(pointer: coarse)").matches;
+		$tw.browser.hasTouch = !!window.matchMedia && window.matchMedia("(pointer: coarse)").matches;
```

Comment:
Recommend refactoring the check for Chrome and Touch as utility functions.

Reply:
I did implement them as 3 new $tw.utils.browser.XXX functions. So it's easy to modify them in a plugin one by one if needed.